ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEventsRangeQueries.scala

```diff
@@ -143,15 +155,33 @@ private[events] object EventsTableFlatEventsRangeQueries {
     implicit def `go by limit`(saferRead: SimpleSql[Row]): ByLimit = ByLimit(saferRead)
   }
 
+  final case class GetTransactionsConfig(includeNonConsumingExerciseEvents: Boolean)
+
   final class GetTransactions(
       selectColumns: String,
       sqlFunctions: SqlFunctions,
-  ) extends EventsTableFlatEventsRangeQueries[EventsRange[Long]] {
+  ) extends EventsTableFlatEventsRangeQueries[EventsRange[Long], GetTransactionsConfig] {
+
+    // this condition allows to include only:
+    // Created events - exercise_consuming is NULL
+    // Archived events - exercise_consuming is TRUE
+    private val FilterOutNonConsumingExerciseEventsWhereClause =
+      """and exercise_consuming IS NOT FALSE"""
+
+    private def filterOutNonConsumingExerciseEventsWhereClause(
+        config: Option[GetTransactionsConfig]
+    ) =
+      config
+        .collect {
+          case GetTransactionsConfig(include) if include => ""
+        }
+        .getOrElse(FilterOutNonConsumingExerciseEventsWhereClause)
```

Comment:
Until now non-consuming events always had `flat_event_witnesses` set empty. This provided filtering out non-consuming events in the write path.
With this PR we intend to fill `flat_event_witnesses` column for both consuming and non-consuming exercise events with appropriate witnesses. This change requires introducing additional filtering to make the switch for including non-consuming events work as intended. To achieve that we add the `FilterOutNonConsumingExerciseEventsWhereClause` `WHERE` clause filter.